integrations/mcp/src/haystack_integrations/tools/mcp/mcp_tool.py

```diff
@@ -914,6 +922,14 @@ def __init__(
         :param eager_connect: If True, connect to server during initialization.
                              If False (default), defer connection until warm_up or first tool use,
                              whichever comes first.
+        :param outputs_to_string: Optional dictionary defining how tool outputs should be converted into a string.
+                                 If the source is provided only the specified output key is sent to the handler.
+                                 If the source is omitted the whole tool result is sent to the handler.
+        :param inputs_from_state: Optional dictionary mapping state keys to tool parameter names.
+                                 Example: `{"repository": "repo"}` maps state's "repository" to tool's "repo" parameter.
+        :param outputs_to_state: Optional dictionary defining how tool outputs map to keys within state as well as
+                                optional handlers. If the source is provided only the specified output key is sent
+                                to the handler.
```

Comment:
Same here let's add an example of how the dict looks 

        Example:
        ```python
        {
            "documents": {"source": "docs", "handler": custom_handler}
        }
        ```
        If the source is omitted the whole tool result is sent to the handler.
        Example:
        ```python
        {
            "documents": {"handler": custom_handler}
        }
        ```